src/extension/tests/Test_ActionHandler.py

```diff
@@ -252,6 +253,9 @@ def test_update_command_fail(self):
         # Remove the directory after the test
         shutil.rmtree(test_dir)
 
+    def test_telemetry_available(self):
+        self.assertTrue(self.runtime.telemetry_writer.is_agent_compatible())
+
     def test_telemetry_available_env_var_not_exists(self):
         # agent env var is not set so telemetry is not supported
```

Comment:
This test looks like is one for telemetry not supported scenario, but the assert on line 277 uses telemetry supported check

self.assertTrue(Constants.TELEMETRY_AT_AGENT_COMPATIBLE_MSG in file_contents)

Reply:
Same comment for the test test_telemetry_available_env_var_key_not_exists

Reply:
You are right that this is not correct. I think that it was added like this because _previously_ we had the strict check (env var and events folder) and it was failing, so those changes were reverted and these tests were changed so that it checks for telemetry existing even though the env var is not there (but events folder is). 

I can change the code to rely on both events folder and env var (strict check).